warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -672,6 +678,17 @@ public static void validateQuerySize(String lastOperation, final JexlNode queryT
             throw new DatawaveFatalQueryException(qe);
         }
 
+        // now check whether we are over the ivarator limit
+        if (maxIvaratorThreshold >= 0) {
+            NodeTypeCount nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
+            int totalIvarators = nodeCount.getTotal(ExceededValueThresholdMarkerJexlNode.class) + nodeCount.getTotal(ExceededOrThresholdMarkerJexlNode.class);
```

Comment:
Seems like the NodeTypeCountVisitor should optionally accept a list of class names as a filter..

Reply:
The visitor still needs to visit every node.  To add an additional check at every node to determine whether it is something we are interested in actually may be more expensive, but that is merely a hypothesis.  I will try to test that out.

Reply:
Well, after much testing it appears that adding a set to test against is neglegible in terms of addition time.  Adding that in.